### ISSUE TITLE
Bindings don't work anymore in Python 3.8

### ISSUE THREAD

**Issue by felixchenier** [pyomeca/ezc3d#282]
Hi

I updated ezc3d on my python 3.8 installation but from ezc3d 1.5.1, the module doesn't load anymore.

I tried to install it on a fresh python installation. It works well on python 3.11, also on python 3.9. But here is the output on a fresh python 3.8 install on a conda distribution:

```
(base) felix@asterix ~ % conda create -n test python=3.8 ezc3d -c conda-forge

Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 22.9.0
  latest version: 22.11.1

Please update conda by running

    $ conda update -n base -c conda-forge conda



## Package Plan ##

  environment location: /Users/felix/opt/miniconda3/envs/test

  added / updated specs:
    - ezc3d
    - python=3.8


The following NEW packages will be INSTALLED:

  arpack             conda-forge/osx-64::arpack-3.7.0-hefb7bc6_2 None
  bzip2              conda-forge/osx-64::bzip2-1.0.8-h0d85af4_4 None
  c-ares             conda-forge/osx-64::c-ares-1.18.1-h0d85af4_0 None
  ca-certificates    conda-forge/osx-64::ca-certificates-2022.12.7-h033912b_0 None
  cairo              conda-forge/osx-64::cairo-1.16.0-h904041c_1014 None
  curl               conda-forge/osx-64::curl-7.87.0-h6df9250_0 None
  expat              conda-forge/osx-64::expat-2.5.0-hf0c8a7f_0 None
  ezc3d              conda-forge/osx-64::ezc3d-1.5.1-py310_octave_h9868650_4 None
  fftw               conda-forge/osx-64::fftw-3.3.10-nompi_h4fa670e_106 None
  fltk               conda-forge/osx-64::fltk-1.3.8-h7ffa276_0 None
  font-ttf-dejavu-s~ conda-forge/noarch::font-ttf-dejavu-sans-mono-2.37-hab24e00_0 None
  font-ttf-inconsol~ conda-forge/noarch::font-ttf-inconsolata-3.000-h77eed37_0 None
  font-ttf-source-c~ conda-forge/noarch::font-ttf-source-code-pro-2.038-h77eed37_0 None
  font-ttf-ubuntu    conda-forge/noarch::font-ttf-ubuntu-0.83-hab24e00_0 None
  fontconfig         conda-forge/osx-64::fontconfig-2.14.1-h5bb23bf_0 None
  fonts-conda-ecosy~ conda-forge/noarch::fonts-conda-ecosystem-1-0 None
  fonts-conda-forge  conda-forge/noarch::fonts-conda-forge-1-0 None
  freetype           conda-forge/osx-64::freetype-2.12.1-h3f81eb7_1 None
  fribidi            conda-forge/osx-64::fribidi-1.0.10-hbcb3906_0 None
  gettext            conda-forge/osx-64::gettext-0.21.1-h8a4c099_0 None
  ghostscript        conda-forge/osx-64::ghostscript-9.54.0-h96cf925_2 None
  giflib             conda-forge/osx-64::giflib-5.2.1-hbcb3906_2 None
  gl2ps              conda-forge/osx-64::gl2ps-1.4.2-h4cff582_0 None
  glib               conda-forge/osx-64::glib-2.74.1-hbc0c0cd_1 None
  glib-tools         conda-forge/osx-64::glib-tools-2.74.1-hbc0c0cd_1 None
  glpk               conda-forge/osx-64::glpk-5.0-h3cb5acd_0 None
  gmp                conda-forge/osx-64::gmp-6.2.1-h2e338ed_0 None
  gnuplot            conda-forge/osx-64::gnuplot-5.4.5-h62d52a3_1 None
  graphite2          conda-forge/osx-64::graphite2-1.3.13-h2e338ed_1001 None
  gst-plugins-base   conda-forge/osx-64::gst-plugins-base-1.21.3-h37e1711_1 None
  gstreamer          conda-forge/osx-64::gstreamer-1.21.3-h1d18e73_1 None
  harfbuzz           conda-forge/osx-64::harfbuzz-6.0.0-h08f8713_0 None
  hdf5               conda-forge/osx-64::hdf5-1.12.2-nompi_h48135f9_101 None
  icu                conda-forge/osx-64::icu-70.1-h96cf925_0 None
  jpeg               conda-forge/osx-64::jpeg-9e-hac89ed1_2 None
  krb5               conda-forge/osx-64::krb5-1.20.1-h049b76e_0 None
  lame               conda-forge/osx-64::lame-3.100-hb7f2c08_1003 None
  lerc               conda-forge/osx-64::lerc-4.0.0-hb486fe8_0 None
  libaec             conda-forge/osx-64::libaec-1.0.6-hf0c8a7f_1 None
  libblas            conda-forge/osx-64::libblas-3.9.0-16_osx64_openblas None
  libcblas           conda-forge/osx-64::libcblas-3.9.0-16_osx64_openblas None
  libclang           conda-forge/osx-64::libclang-13.0.1-default_he082bbe_0 None
  libcurl            conda-forge/osx-64::libcurl-7.87.0-h6df9250_0 None
  libcxx             conda-forge/osx-64::libcxx-14.0.6-hccf4f1f_0 None
  libdeflate         conda-forge/osx-64::libdeflate-1.17-hac1461d_0 None
  libedit            conda-forge/osx-64::libedit-3.1.20191231-h0678c8f_2 None
  libev              conda-forge/osx-64::libev-4.33-haf1e3a3_1 None
  libffi             conda-forge/osx-64::libffi-3.4.2-h0d85af4_5 None
  libflac            conda-forge/osx-64::libflac-1.4.2-hf0c8a7f_0 None
  libgd              conda-forge/osx-64::libgd-2.3.3-h4685329_4 None
  libgfortran        conda-forge/osx-64::libgfortran-5.0.0-11_3_0_h97931a8_27 None
  libgfortran5       conda-forge/osx-64::libgfortran5-11.3.0-h082f757_27 None
  libglib            conda-forge/osx-64::libglib-2.74.1-h4c723e1_1 None
  libiconv           conda-forge/osx-64::libiconv-1.17-hac89ed1_0 None
  liblapack          conda-forge/osx-64::liblapack-3.9.0-16_osx64_openblas None
  liblapacke         conda-forge/osx-64::liblapacke-3.9.0-16_osx64_openblas None
  libllvm13          conda-forge/osx-64::libllvm13-13.0.1-h64f94b2_2 None
  libnghttp2         conda-forge/osx-64::libnghttp2-1.51.0-he2ab024_0 None
  libogg             conda-forge/osx-64::libogg-1.3.4-h35c211d_1 None
  libopenblas        conda-forge/osx-64::libopenblas-0.3.21-openmp_h429af6e_3 None
  libopus            conda-forge/osx-64::libopus-1.3.1-hc929b4f_1 None
  libpng             conda-forge/osx-64::libpng-1.6.39-ha978bb4_0 None
  libpq              conda-forge/osx-64::libpq-15.1-h3640bf0_3 None
  libsndfile         conda-forge/osx-64::libsndfile-1.2.0-h591af1c_0 None
  libsqlite          conda-forge/osx-64::libsqlite-3.40.0-ha978bb4_0 None
  libssh2            conda-forge/osx-64::libssh2-1.10.0-h47af595_3 None
  libtiff            conda-forge/osx-64::libtiff-4.5.0-hee9004a_2 None
  libvorbis          conda-forge/osx-64::libvorbis-1.3.7-h046ec9c_0 None
  libwebp            conda-forge/osx-64::libwebp-1.2.4-h70a068d_1 None
  libwebp-base       conda-forge/osx-64::libwebp-base-1.2.4-h775f41a_0 None
  libxcb             conda-forge/osx-64::libxcb-1.13-h0d85af4_1004 None
  libxml2            conda-forge/osx-64::libxml2-2.10.3-hb9e07b5_0 None
  libzlib            conda-forge/osx-64::libzlib-1.2.13-hfd90126_4 None
  llvm-openmp        conda-forge/osx-64::llvm-openmp-15.0.7-h61d9ccf_0 None
  metis              conda-forge/osx-64::metis-5.1.0-h2e338ed_1006 None
  mpfr               conda-forge/osx-64::mpfr-4.1.0-h0f52abe_1 None
  mpg123             conda-forge/osx-64::mpg123-1.31.2-hf0c8a7f_0 None
  mysql-common       conda-forge/osx-64::mysql-common-8.0.31-hc4b2c72_0 None
  mysql-libs         conda-forge/osx-64::mysql-libs-8.0.31-h8658499_0 None
  ncurses            conda-forge/osx-64::ncurses-6.3-h96cf925_1 None
  nspr               conda-forge/osx-64::nspr-4.35-hea0b92c_0 None
  nss                conda-forge/osx-64::nss-3.78-ha8197d3_0 None
  octave             conda-forge/osx-64::octave-7.3.0-h68757aa_3 None
  openssl            conda-forge/osx-64::openssl-3.0.7-hfd90126_1 None
  packaging          conda-forge/noarch::packaging-23.0-pyhd8ed1ab_0 None
  pango              conda-forge/osx-64::pango-1.50.12-hbd9bf65_1 None
  pcre               conda-forge/osx-64::pcre-8.45-he49afe7_0 None
  pcre2              conda-forge/osx-64::pcre2-10.40-h1c4e4bc_0 None
  perl               conda-forge/osx-64::perl-5.32.1-2_h0d85af4_perl5 None
  pip                conda-forge/noarch::pip-22.3.1-pyhd8ed1ab_0 None
  pixman             conda-forge/osx-64::pixman-0.40.0-hbcb3906_0 None
  ply                conda-forge/noarch::ply-3.11-py_1 None
  portaudio          conda-forge/osx-64::portaudio-19.6.0-hf0c8a7f_7 None
  pthread-stubs      conda-forge/osx-64::pthread-stubs-0.4-hc929b4f_1001 None
  pyqt               conda-forge/osx-64::pyqt-5.15.7-py38h872f513_2 None
  pyqt5-sip          conda-forge/osx-64::pyqt5-sip-12.11.0-py38haf9be31_2 None
  python             conda-forge/osx-64::python-3.8.15-hf9b03c3_0_cpython None
  python_abi         conda-forge/osx-64::python_abi-3.8-3_cp38 None
  qhull              conda-forge/osx-64::qhull-2020.2-h940c156_2 None
  qscintilla2        conda-forge/osx-64::qscintilla2-2.13.3-py38h903b8e1_4 None
  qt-main            conda-forge/osx-64::qt-main-5.15.6-haeff654_5 None
  readline           conda-forge/osx-64::readline-8.1.2-h3899abd_0 None
  setuptools         conda-forge/noarch::setuptools-66.0.0-pyhd8ed1ab_0 None
  sip                conda-forge/osx-64::sip-6.7.5-py38h4cd09af_0 None
  sqlite             conda-forge/osx-64::sqlite-3.40.0-h9ae0607_0 None
  suitesparse        conda-forge/osx-64::suitesparse-5.10.1-h7aff33d_1 None
  sundials           conda-forge/osx-64::sundials-6.5.0-h4d39bc9_0 None
  tbb                conda-forge/osx-64::tbb-2021.7.0-hb8565cd_1 None
  texinfo            conda-forge/osx-64::texinfo-7.0-pl5321hc47821c_0 None
  tk                 conda-forge/osx-64::tk-8.6.12-h5dbffcc_0 None
  toml               conda-forge/noarch::toml-0.10.2-pyhd8ed1ab_0 None
  wheel              conda-forge/noarch::wheel-0.38.4-pyhd8ed1ab_0 None
  xorg-fixesproto    conda-forge/osx-64::xorg-fixesproto-5.0-h0d85af4_1002 None
  xorg-kbproto       conda-forge/osx-64::xorg-kbproto-1.0.7-h35c211d_1002 None
  xorg-libice        conda-forge/osx-64::xorg-libice-1.0.10-h0d85af4_0 None
  xorg-libsm         conda-forge/osx-64::xorg-libsm-1.2.3-h0d85af4_1000 None
  xorg-libx11        conda-forge/osx-64::xorg-libx11-1.7.2-h0d85af4_0 None
  xorg-libxau        conda-forge/osx-64::xorg-libxau-1.0.9-h35c211d_0 None
  xorg-libxdmcp      conda-forge/osx-64::xorg-libxdmcp-1.1.3-h35c211d_0 None
  xorg-libxext       conda-forge/osx-64::xorg-libxext-1.3.4-h0d85af4_1 None
  xorg-libxfixes     conda-forge/osx-64::xorg-libxfixes-5.0.3-h0d85af4_1004 None
  xorg-libxrender    conda-forge/osx-64::xorg-libxrender-0.9.10-h0d85af4_1003 None
  xorg-renderproto   conda-forge/osx-64::xorg-renderproto-0.11.1-h0d85af4_1002 None
  xorg-xextproto     conda-forge/osx-64::xorg-xextproto-7.3.0-h35c211d_1002 None
  xorg-xproto        conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007 None
  xz                 conda-forge/osx-64::xz-5.2.6-h775f41a_0 None
  zlib               conda-forge/osx-64::zlib-1.2.13-hfd90126_4 None
  zstd               conda-forge/osx-64::zstd-1.5.2-hfa58983_4 None


Proceed ([y]/n)? 

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate test
#
# To deactivate an active environment, use
#
#     $ conda deactivate

Retrieving notices: ...working... done
```

```
(base) felix@asterix ~ % conda activate test
(test) felix@asterix ~ % python 
Python 3.8.15 | packaged by conda-forge | (default, Nov 22 2022, 08:53:40) 
[Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ezc3d
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'ezc3d'
>>> 
```

Is it a bug, or is python 3.8 not supported anymore? I think it would be a good idea to clarify, so that users (like me) can clarify the dependencies.

Thanks!

Félix

**Comment by pariterre** [comment]
Hello @felixchenier 

Python 3.8 and Python 3.9 are indeed not supported anymore. This is mainly because of the way hinting in function are written since Python 3.10

**Comment by felixchenier** [comment]
@pariterre OK thanks for this ultra-fast response. I understand the idea. I tried to support older Python version in ktk so I refrained from using the nice annotation stuff in Python 3.9 and 3.10, but now I have a good reason to bump this requirement. I don't know if I should be angry or thank you instead 😁. Anyway I tag you as the direct cause for this bump!! ;-)

PS I suggest however to add a python>=3.10 in your conda recipe. I'll do the same for mine.
